tests/test_api_v1.py

```diff
@@ -505,6 +506,100 @@ def test_create_request_filter_state(app, auth_env, client, db):
         assert fetched_requests[0]["state"] == state
 
 
+def test_create_request_filter_date(app, auth_env, client, db):
+    """Test that requests can be filtered by date."""
+    # flask_login.current_user is used in Request.from_json, which requires a request context
+    with app.test_request_context(environ_base=auth_env):
+        # Make a request with earliest creation date
+        first_date = "2020-01-01"
+        data_first = {
+            "created": date.fromisoformat(first_date),
+            "repo": "https://github.com/release-engineering/retrodep.git",
+            "ref": "c50b93a32df1c9d700e3e80996845bc2e13be848",
+            "pkg_managers": ["gomod"],
+        }
+        request_first = Request.from_json(data_first)
+        db.session.add(request_first)
+        # Make a request with latest creation date
+        last_date = "2021-01-01"
+        data_last = {
+            "created": date.fromisoformat(last_date),
+            "repo": "https://github.com/release-engineering/retrodep.git",
+            "ref": "12a0692be09fc18ce82a71904562d8408fe2296a",
+            "pkg_managers": ["gomod"],
+        }
+        request_last = Request.from_json(data_last)
+        db.session.add(request_last)
+    db.session.commit()
+
+    test_date = "2020-06-30"
+    # Test created_from parameter
+    rv = client.get(f"/api/v1/requests?created_from={test_date}")
+    assert rv.status_code == 200
+    fetched_requests = rv.json["items"]
+    assert len(fetched_requests) == 1
+    assert fetched_requests[0]["ref"] == "12a0692be09fc18ce82a71904562d8408fe2296a"
+    # Test created_to parameter
+    rv = client.get(f"/api/v1/requests?created_to={test_date}")
+    assert rv.status_code == 200
+    fetched_requests = rv.json["items"]
+    assert len(fetched_requests) == 1
+    assert fetched_requests[0]["ref"] == "c50b93a32df1c9d700e3e80996845bc2e13be848"
+    # Test created_from and created_to together
+    rv = client.get(f"/api/v1/requests?created_from={last_date}&created_to={last_date}")
+    assert rv.status_code == 200
+    fetched_requests = rv.json["items"]
+    assert len(fetched_requests) == 1
+    assert fetched_requests[0]["ref"] == "12a0692be09fc18ce82a71904562d8408fe2296a"
+
+
+def test_create_request_filter_datetime(app, auth_env, client, db):
```

Comment:
The test looks like previous one with different input.
I would suggest replace those two tests with one, parametrize it and test more cases.
Example:
```python
@pytest.mark.parametrize("created_list,created_filter,expected_num", [
    (["2021-01-01", "2021-02-02"], {"created_from": "2021-01-01", "created_to": "2021-02-02"}, 2),
    ...
])
def test_requests_created_filter(..., created_list, created_filter, expected_num):
    ...
    for created in created_list:
        data = {
            "created": date.fromisoformat(created),
            ...
        }
        ...
    rv = client.get(f"/api/v1/requests?{urlencode(created_filter)}")
    ...
    assert len(fetched_requests) == expected_num
```

You can add as many parameters as needed.

Reply:
Done!